oracle_fdw.c

```diff
@@ -821,9 +822,63 @@ oracleGetForeignRelSize(PlannerInfo *root, RelOptInfo *baserel, Oid foreigntable
 				fdwState->oraTable->cols[i]->used = 1;
 	}
 
-	/* construct Oracle query and get the list of parameters and actions for RestrictInfos */
-	fdwState->query = createQuery(fdwState, baserel, for_update, root->query_pathkeys);
-	elog(DEBUG1, "oracle_fdw: remote query is: %s", fdwState->query);
+	/* createQuery is moved to GetForeignPlan. */
+	/* fdwState->for_update is needed by createQuery in GetForeignPlans */
+	fdwState->for_update = for_update;
+	/* fdwState->usable_pathkeys is needed by create_foreign_scan in GetForeignPaths. */
+	/* So it is created in this place. */
```

Comment:
That doesn't seem to make much sense.
If `usable_pathkeys` is needed in `oracleGetForeignPaths`, this whole block should be moved there.
Then `usable_pathkeys` would not be a member in `fdwState`, but a local variable in `oracleGetForeignPaths`.